### ISSUE TITLE
Duplicate executors created when multiple jobs are configured and executed

### ISSUE THREAD

**Issue by jewzaam** [lightblue-platform/lightblue-migrator#124]
Logs for migrator shoing the behavior:
```
2015-03-23 17:42:24,381 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - Loading jobs for entity1
2015-03-23 17:42:26,555 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - Loaded jobs for entity1: 10
2015-03-23 17:42:26,556 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - Loading jobs for entity2
2015-03-23 17:42:50,317 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - Loaded jobs for entity2: 80689
2015-03-23 17:42:56,952 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - Done setting up job executors
2015-03-23 17:48:49,179 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - Job executors done
2015-03-23 17:48:50,672 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - Loading jobs for entity1
2015-03-23 17:48:52,355 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - Loaded jobs for entity1: 0
2015-03-23 17:48:52,355 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - Loading jobs for entity2
2015-03-23 17:49:15,156 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - Loaded jobs for entity2: 77923
2015-03-23 17:49:20,137 [ConsistencyCheckerRunner] com.redhat.lightblue.migrator.consistency.ConsistencyChecker INFO - Done setting up job executors
```

DB in mongo showing multiple jobExecutions:
```
> db.migrationJob.find({"jobExecutions#":{$gt:1}}).count()
1408
```

We're running one instance of the migrator, so shouldn't have any cases of more than one execution happening on any one job!

**Comment by jewzaam** [comment]
This is because `MAX_THREAD_WAIT_MSEC` is set to 5 minutes.  This means that after executor setup each executor has 5 minutes to complete before it is terminated.

https://github.com/lightblue-platform/lightblue-migrator/blob/2abd7c39a24deb21a964c5c524658ff6cef7d048/consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/ConsistencyChecker.java#L185

Good news: the threads are not running parallel.
Bad news: this means possibly many threads in flight are terminated early.

Possible fixes:
* increase timeout

**Comment by jewzaam** [comment]
oops...

Possible fixes:
1. increase timeout
2. update job pulling to pull a smaller set of jobs
3. revisit what happens after executors are setup

I'd like to look at the third option.  Right now all jobs are setup and then almost immediately marked for shutdown.  This doesn't seem very efficient.  Or, I'm not reading how this works correctly.

**Comment by jewzaam** [comment]
Yea, I was reading it wrong.  So, each set of entities being migrated has a single executor in which all jobs are executed.  Calling shutdown on that executor will allow all submitted jobs to process.  Calling awaitTermination waits for a period of time for jobs to complete.  Therefore option 3 is not going to work.

Increasing the timeout is the easiest thing to do.  We know the migration we ran took less than 6 hrs for 10 million records.  This is a pretty representative size for migrations we'll have in the near future.  If we set the timeout to something long, 12 hours, it would cover this.

Note that pulling fewer jobs would also help, reducing the working set and meaning a smaller timeout could be used.  Immediately after completing work the migrator looks for new work, so this would incur a minimal overhead.
